### DISCUSSION TITLE
[BUG] Allow MemberCanShare doesn't work when Authentication with Credentials or with Multi-Factor authentication enabled

### DISCUSSION THREAD

**Discussion by ibaettigdev** [pnp/powershell#3519] [Q&A]
### Reporting an Issue or Missing Feature
I try to enable sharing for members, but i always get an Access denied error.
My Account is Global Admin and also SiteAdministrator of the SPO Site
Connect-PnPOnline "https://tenantname.sharepoint.com/sites/testsite" -interactive
If i authenticate with an Azure AD App Registration it works 
Connect-PnPOnline "https://tenantname.sharepoint.com/sites/testsite" -ClientId $clientid -Thumbprint $thumbprint -Tenant $tenantid

### Expected behavior 
It should change the SharingSetting to SiteMembers can share files and folders also with interactive login

### Actual behavior 
```
Connect-PnPOnline "https://tenantname.sharepoint.com/sites/testsite" -interactive
$web = Get-PnPWeb -Includes MembersCanShare, AssociatedMemberGroup.AllowMembersEditMembership
$web.MembersCanShare = $true
$web.Update()
$web.Context.ExecuteQuery()

MethodInvocationException: 
Line |
   5 |  $web.Context.ExecuteQuery()
     |  ~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Exception calling "ExecuteQuery" with "0" argument(s): "Access is denied. (Exception from HRESULT: 0x80070005 (E_ACCESSDENIED))"
```

### Steps to reproduce behavior
if you run this code it produces an error:
```
Connect-PnPOnline "https://tenantname.sharepoint.com/sites/testsite" -interactive
$web = Get-PnPWeb -Includes MembersCanShare, AssociatedMemberGroup.AllowMembersEditMembership
$web.MembersCanShare = $true
$web.Update()
$web.Context.ExecuteQuery()
```
if you run this code it works:
```
Connect-PnPOnline "https://tenantname.sharepoint.com/sites/testsite" -ClientId $clientid -Thumbprint $thumbprint -Tenant $tenantid
$web = Get-PnPWeb -Includes MembersCanShare, AssociatedMemberGroup.AllowMembersEditMembership
$web.MembersCanShare = $true
$web.Update()
$web.Context.ExecuteQuery()
```

### What is the version of the Cmdlet module you are running?
PnP.Powershell 2.2.0

### Which operating system/environment are you running PnP PowerShell on?
- [x] Windows
- [ ] Linux
- [ ] MacOS
- [ ] Azure Cloud Shell
- [ ] Azure Functions
- [ ] Other : please specify

**Comment by ibaettigdev** [comment]
additional information:
`Set-PnPTenantSite -Identity "https://tenantname.sharepoint.com/sites/testsite" -DisableSharingForNonOwners:$false`
or 
`Set-PnPSite -DisableSharingForNonOwners:$false`
does not work regardless of authentication

**Comment by ibaettigdev** [comment]
also the same behavior when i use the Parameter "MembersCanShare="true" in the pnp provisioning template 
works with app registration --  doesnt work with -interactive authentication
`<pnp:WebSettings NoCrawl="false" WelcomePage="SitePages/Home.aspx" CommentsOnSitePagesDisabled="false" QuickLaunchEnabled="false" SearchScope="DefaultScope" SearchBoxInNavBar="Inherit" MembersCanShare="true" />`

**Comment by Micael-stack** [comment]
Not a bug anymore since -interactive is mendatory with -clientId in PNP since a couple of versions ago. So no more problems.
